src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/AnalyzePackageStatus.cs

```diff
@@ -101,7 +101,15 @@ void GetResults<T>(string name, string action, IReadOnlyCollection<T> collection
                     {
                         foreach (var s in collection)
                         {
-                            results.Add(string.Concat(name, s, action));
+                            var mc = s as NuGetReference;
```

Comment:
Why the type check here? If you do need it, better to switch to pattern matching